src/hotspot/share/gc/shenandoah/heuristics/shenandoahOldHeuristics.cpp

```diff
@@ -546,40 +546,48 @@ void ShenandoahOldHeuristics::clear_triggers() {
   _growth_trigger = false;
 }
 
-void ShenandoahOldHeuristics::trigger_collection_if_fragmented(size_t first_old_region, size_t last_old_region, size_t old_region_count, size_t num_regions) {
+// This triggers old-gen collection if the number of regions "dedicated" to old generation is much larger than
+// is required to represent the memory currently used within the old generation.  This trigger looks specifically
+// at density of the old-gen spanned region.  A different mechanism triggers old-gen GC if the total number of
+// old-gen regions (regardless of how close the regions are to one another) grows beyond an anticipated growth target.
```

Comment:
I realize this is a private API / work method. Would still be good to place the documentation comment in the .hpp file.

```
// If the old generation is fragmented, set appropriate collection triggering signals.
```

Reply:
In this implementation, I'd then describe how the fragmentation is computed.

Reply:
Also, I'd avoid using "trigger", and instead say "set_trigger", since the actual collection will happen later, as far as I understand.

Reply:
Thanks.  Adding comment and changing name to set_trigger_if_old_is_fragmented().  Also, renaming trigger_collection_if_overgown() to set_trigger_if_old_is_overgrown().